src/main/java/com/ibm/nfvodriver/service/LifecycleManagementService.java

```diff
@@ -40,77 +47,60 @@ public LifecycleManagementService(NSLifecycleManagementDriver nsLifecycleManagem
+                    return new ExecutionAcceptedResponse(responseGetInstantiateNsUUID);
+                case LIFECYCLE_UPGRADE:
+                    // Upgrade
+                    final String nsUpgradeInstanceId = executionRequest.getStringResourceProperty("nsInstanceId");
+                    final String responseGetUpdateNsUUID = nsLifecycleManagementDriver.updateNs(executionRequest.getDeploymentLocation(), nsUpgradeInstanceId, requestPayload);
+                    return new ExecutionAcceptedResponse(responseGetUpdateNsUUID);
+                case LIFECYCLE_DELETE:
+                    // Delete
+                    final String nsInstanceId = executionRequest.getStringResourceProperty("nsInstanceId");
+                    nsLifecycleManagementDriver.deleteNsInstance(executionRequest.getDeploymentLocation(), nsInstanceId, requestId);
+                    externalMessagingService.sendDelayedExecutionAsyncResponse(new ExecutionAsyncResponse(requestId, ExecutionStatus.COMPLETE, null, Collections.emptyMap(), Collections.emptyMap()), properties.getExecutionResponseDelay());
+                    return new ExecutionAcceptedResponse(requestId);
+                case LIFECYCLE_UNINSTALL:
+                    // Terminate
+                    final String nsUninstallInstanceId = executionRequest.getStringResourceProperty("nsInstanceId");
+                    final String responseGetTerminateNsUUID = nsLifecycleManagementDriver.terminateNs(executionRequest.getDeploymentLocation(), nsUninstallInstanceId, requestPayload);
+                    return new ExecutionAcceptedResponse(responseGetTerminateNsUUID);
+                case LIFECYCLE_SCALEIN:
+                case LIFECYCLE_SCALEOUT:
+                case LIFECYCLE_SCALETOLEVEL:
+                    // ScaleIn/ScaleOut/ScaleToLevel
+                    final String nsScaleInstanceId = executionRequest.getStringResourceProperty("nsInstanceId");
+                    final String responseGetScaleNsUUID = nsLifecycleManagementDriver.scaleNs(executionRequest.getDeploymentLocation(), nsScaleInstanceId, requestPayload);
+                    return new ExecutionAcceptedResponse(responseGetScaleNsUUID);
+                case LIFECYCLE_HEAL:
+                    // Heal
+                    final String nsHealInstanceId = executionRequest.getStringResourceProperty("nsInstanceId");
+                    final String responseGetHealNsUUID = nsLifecycleManagementDriver.healNs(executionRequest.getDeploymentLocation(), nsHealInstanceId, requestPayload);
+                    return new ExecutionAcceptedResponse(responseGetHealNsUUID);
+                default:
+                    throw new IllegalArgumentException(String.format("Requested transition [%s] is not supported by this lifecycle driver", executionRequest.getLifecycleName()));
```

Comment:
As this exception is already handled in else block above, This can be a driver error.